uberenv.py

```diff
@@ -320,13 +347,123 @@ def detect_platform(self):
         return res
 
 
+class VcpkgEnv(UberEnv):
+    """ Helper to clone vcpkg and install libraries on Windows """
+
+    def __init__(self, opts, extra_opts):
+        UberEnv.__init__(self,opts,extra_opts)
+
+        # setup architecture triplet
+        self.vcpkg_triplet = self.set_from_args_or_json("vcpkg_triplet")
+        print("Vcpkg triplet: {}".format(self.vcpkg_triplet))
+        if self.vcpkg_triplet is None:
+           self.vcpkg_triplet = os.getenv("VCPKG_DEFAULT_TRIPLET", "x86-windows")
+
+    def setup_paths_and_dirs(self):
+        # get the current working path, and the glob used to identify the
+        # package files we want to hot-copy to vcpkg
+
+        UberEnv.setup_paths_and_dirs(self)
+
+        self.ports = pjoin(self.uberenv_path, "vcpkg_ports","*")
+
+        # setup path for vcpkg repo
+        print("[installing to: {0}]".format(self.dest_dir))
+        self.dest_vcpkg = pjoin(self.dest_dir,"vcpkg")
+
+        if os.path.isdir(self.dest_vcpkg):
+            print("[info: destination '{0}' already exists]".format(self.dest_vcpkg))
+
+    def clone_repo(self):
+        if not os.path.isdir(self.dest_vcpkg):
+            # compose clone command for the dest path, vcpkg url and branch
+            vcpkg_branch = self.project_opts.get("vcpkg_branch", "master")
+            vcpkg_url = self.project_opts.get("vcpkg_url", "https://github.com/microsoft/vcpkg")
+
+            print("[info: cloning vcpkg '{0}' branch from {1} into {2}]"
+                .format(vcpkg_branch,vcpkg_url, self.dest_vcpkg))
+
+            os.chdir(self.dest_dir)
+
+            clone_opts = ("-c http.sslVerify=false " 
+                          if self.opts["ignore_ssl_errors"] else "")
+
+            clone_cmd =  "git {0} clone -b {1} {2} vcpkg".format(clone_opts, vcpkg_branch,vcpkg_url)
+            sexe(clone_cmd, echo=True)
+
+            # optionally, check out a specific commit
+            if "vcpkg_commit" in self.project_opts:
+                sha1 = self.project_opts["vcpkg_commit"]
+                print("[info: using vcpkg commit {0}]".format(sha1))
+                os.chdir(self.dest_vcpkg)
+                sexe("git checkout {0}".format(sha1),echo=True)
```

Comment:
Warning: The assumption here is that the commit is in the history of the fetched branch.
E.g., the default branch being `master` it is not possible to ask for a commit more recent without also specifying a branch that contains it.
In Spack case, I use fetch to prevent this.

Reply:
I'll make this consistent with `spack`'s `repo_pull` which appears to error out if the commit is not present